src/lein_monolith/task/checkouts.clj

```diff
@@ -57,16 +57,26 @@
 (defn link
   "Create symlinks in the checkouts directory pointing to all internal
   dependencies in the current project."
-  [project opts]
+  [project opts project-names]
   (let [[monolith subprojects] (u/load-monolith! project)
         dep-map (dep/dependency-map subprojects)
-        projects-to-link (as-> (:dependencies project) deps
-                           (map (comp dep/condense-name first) deps)
-                           (if (:deep opts)
-                             (set (mapcat (partial dep/upstream-keys dep-map) deps))
-                             deps)
-                           (keep subprojects deps))
+        selected-names (into #{}
+                             (map (partial dep/resolve-name! (keys dep-map)))
+                             project-names)
+        projects-to-link (cond->> (map (comp dep/condense-name first)
+                                       (:dependencies project))
+                           (or (:deep opts) (seq project-names))
+                             (mapcat (partial dep/upstream-keys dep-map))
+                           true
+                             (distinct)
+                           (seq selected-names)
+                             (filter selected-names)
+                           true
+                             (keep subprojects))
         checkouts-dir (io/file (:root project) "checkouts")]
+    (when (empty? projects-to-link)
+      (lein/abort (str "Couldn't find any projects to link matching: "
+                       (str/join " " project-names))))
```

Comment:
When would `projects-to-link` be empty, but one of the calls to `dep/resolve-name!` didn't already throw an exception?

Reply:
When you specify a valid subproject name, but it is not in the upstream dependency closure of the project you are in.